src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ImportStatusBauService.cs

```diff
@@ -48,11 +49,30 @@ public class ImportStatusBauService : IImportStatusBauService
 
                     return queryResults.Where(result =>
                     {
-                        if (result.Message == "")
+                        try
+                        {
+                            var m = JsonConvert.DeserializeObject<ImportMessage>(result.Message);
+
+                            if (m.Release?.Id == null || m.Release?.Publication?.Id == null || m.MetaFileName == null)
+                            {
+                                throw new NullReferenceException();
+                            }
+                        }
+                        catch (NullReferenceException e)
+                        {
+                            _logger.LogWarning(
+                                $"{MethodBase.GetCurrentMethod()}: Parsed Message contains null values. " +
+                                "These  are needed to create new ImportStatusBau. " +
+                                $"PartitionKey: \"{result.PartitionKey}\", RowKey: \"{result.RowKey}\", " +
+                                $"NullReferenceException: \"{e}\"");
+                            return false;
+                        }
+                        catch (JsonReaderException e)
                         {
                             _logger.LogWarning(
-                                $"{MethodBase.GetCurrentMethod()} received Imports row with empty Message. " +
-                                $"PartitionKey: \"{result.PartitionKey}\", RowKey: \"{result.RowKey}\"");
+                                $"{MethodBase.GetCurrentMethod()}: JsonReaderException when parsing Message. " +
```

Comment:
If either of these warnings are quite common cases then we could work out why and clean up the data especially if they are import rows relating to files that no longer exist. Otherwise this could be quite noisy in App Insights logging.

Reply:
The thought did cross my mind that this might spam warnings - I'll create a separate ticket to clean up bad data on the environments.